### PR TITLE
Use a shallower tree

### PR DESCRIPTION
**Pull request by greg0ire**
This is fuzzy-matcher friendly, and will make our configuration files easier to read.

Blocked by #973 and #974

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -2,58 +2,58 @@ parameters:
...
-        - tests/Doctrine/Tests/Common/Proxy/PHP81IntersectionTypes.php
-        - tests/Doctrine/Tests/Common/Proxy/Php8UnionTypes.php
-        - tests/Doctrine/Tests/Common/Proxy/Php8StaticType.php
-        - tests/Doctrine/Tests/Common/Proxy/ProxyGeneratorTest.php
-        - tests/Doctrine/Tests/Common/Proxy/ProxyLogicTypedPropertiesTest.php
-        - tests/Doctrine/Tests/Common/Proxy/SerializedClass.php
-        - tests/Doctrine/Tests/Common/Proxy/VariadicTypeHintClass.php
-        - tests/Doctrine/Tests/Common/Proxy/Php71NullableDefaultedNonOptionalHintClass.php
-        - tests/Doctrine/Tests/Common/Proxy/generated
+        - src/vendor/doctrine-build-common
```
Comment (stof) [inline phpstan.neon.dist]: what is this path ?

File: psalm.xml [hunk 2]
```diff
@@ -8,11 +8,11 @@
...
 >
     <projectFiles>
-        <directory name="lib/Doctrine/Common" />
+        <directory name="src" />
         <ignoreFiles>
             <!-- Remove ProxyGeneratorTest once Psalm supports native intersection https://github.com/vimeo/psalm/issues/6413 -->
-            <file name="lib/Doctrine/Common/Proxy/ProxyGenerator.php" />
-            <file name="tests/Doctrine/Tests/Common/Proxy/ProxyGeneratorTest.php" />
+            <file name="src/Proxy/ProxyGenerator.php" />
+            <file name="tests/Common/Proxy/ProxyGeneratorTest.php" />
```
Comment (stof) [inline psalm.xml]: why are we exclude files and directories that are not covered by the project files ?

File: phpstan.neon.dist (hunk 1 above)
Comment (greg0ire) [reply phpstan.neon.dist]: I'll make a separate PR to get rid of it, it should have been removed in 4167060592de4dce2b955c6b6dded9a1b43a04b9 I think

Comment (greg0ire) [reply phpstan.neon.dist]: #976

File: psalm.xml (hunk 2 above)
Comment (greg0ire) [reply psalm.xml]: Addressed in #977

**Review by malarzm — APPROVED** [review]

**Pull request merged**